client/src/pages/EditProfile.js

```diff
@@ -408,42 +140,36 @@ function EditProfile() {
         newPipeline.splice(index, 1)
 
         let comp = {
-            id: experience.id,
+            tempId2: experience.tempId2,
+            companyId: experience.companyId,
             name: experience.companyName,
             title: experience.title,
-            removeRating: experience.rating,
+            rating: experience.rating,
             startDate: experience.startDate,
             endDate: experience.endDate,
-            removeEmployees: [user.profileId],
+            userId: user.profileId,
+            indefinite: experience.isIndefinite,
+            remove: true,
         }
 
         let tempOrig = origCompanies
         let found = -1
-        for (let i = 0; i < tempOrig.length; i++) {
-            let company = tempOrig[i]
-            if (comp.id === company.id) {
-                tempOrig.splice(i, 1)
-                found = i
+        let left = 0
```

Comment:
Love the binary search! I think we can make this a function and store it in our `generalUtils.js`.